storage/remote/write_handler.go

```diff
@@ -134,35 +75,59 @@ func NewWriteHandler(logger log.Logger, reg prometheus.Registerer, appendable st
 	return h
 }
 
-func (h *writeHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	var err error
+func (h *writeHandler) parseEncoding(contentEncoding string) (config.RemoteWriteCompression, error) {
+	// TODO(bwplotka): TBD
+	got := config.RemoteWriteCompression(contentEncoding)
+	if err := got.Validate(); err != nil {
+		return "", err // TODO(bwplotka): Wrap properly
+	}
 
-	// Set the header(s) in the response based on the rwFormat the server supports.
-	for hName, hValue := range rwHeaderNameValues(h.rwFormat) {
-		w.Header().Set(hName, hValue)
+	// TODO(bwplotka): Build utils for this.
+	for _, c := range h.compressions {
+		if c == got {
+			return got, nil
+		}
 	}
+	return "", fmt.Errorf("unsupported compression, got %v, supported %v", got, h.acceptEncodingHeaderValue)
+}
 
-	// Parse the headers to work out how to handle this.
-	contentEncoding := r.Header.Get("Content-Encoding")
-	protoVer := r.Header.Get(RemoteWriteVersionHeader)
+func (h *writeHandler) parseType(contentType string) (config.RemoteWriteProtoType, error) {
+	// TODO(bwplotka): TBD
```

Comment:
?

Reply:
I gave up and figured out we need HTTP util pkg for preferences (:  TBD, IF we want to go this way (Accet + Accept-Encoding)